distributions/openhab/src/main/resources/bin/backup.ps1

```diff
@@ -0,0 +1,118 @@
+    Backup-openHAB -OHDirectory C:\openHAB2 -OHBackups c:\openHAB2-backup -ZipFileOut backup.zip
+    #>
+
+    [CmdletBinding()]
+    param(
+        [Parameter(ValueFromPipeline=$True)]
+        [string]$OHDirectory = ".",
+        [Parameter(ValueFromPipeline=$True)]
+        [string]$OHBackups,
+        [Parameter(ValueFromPipeline=$True)]
+        [string]$ZipFileOut
+    )
+
+    begin {}
+    process {
+
+        if (!([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()
+            ).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
+            throw "This script must be run as an Administrator. Start PowerShell with the Run as Administrator option"
+        }
+
+        Write-Host -ForegroundColor Cyan "Checking the specified openHAB directory..."
+        if (!(Test-Path "$OHDirectory\userdata") -And !(Test-Path -Path "$OHDirectory\conf")) {
+            throw "$OHDirectory\userdata doesn't exist! Make sure you are in the " +
+                "openHAB directory or specify the -OHDirectory parameter!"
+        }
+
+        if ($OHDirectory -eq '.') {$OHDirectory = pwd }
+
+        if ([Environment]::GetEnvironmentVariable("OPENHAB_CONF", "Machine")) {
+            $OHConf = [Environment]::GetEnvironmentVariable("OPENHAB_CONF", "Machine")
+        } else {
+            $OHConf = "$OHDirectory\conf"
+        }
+        if ([Environment]::GetEnvironmentVariable("OPENHAB_USERDATA", "Machine")) {
+            $OHUserdata = [Environment]::GetEnvironmentVariable("OPENHAB_USERDATA", "Machine")
+        } else {
+            $OHUserdata = "$OHDirectory\userdata"
+        }
+        if (!($OHBackups)) {
+            if ([Environment]::GetEnvironmentVariable("OPENHAB_BACKUPS", "Machine")) {
+                $OHBackups = [Environment]::GetEnvironmentVariable("OPENHAB_BACKUPS", "Machine")
+            } else {
+                $OHBackups = "$OHDirectory\backups"
+                if (!(Test-Path "$OHDirectory\backups")){
+                    mkdir "$OHBackups" | Out-Null
+                }
+            }
+        }
+
+        Write-Host -ForegroundColor Yellow "Using $OHConf as conf folder"
+        Write-Host -ForegroundColor Yellow "Using $OHUserdata as userdata folder"
+        Write-Host -ForegroundColor Yellow "Using $OHBackups as backups folder"
+
+        $VersionLine = Get-Content "$OHDirectory\userdata\etc\version.properties" | Where-Object { $_.Contains("openhab-distro")}
+        $CurrentVersionIndex = $VersionLine.IndexOf(":")
+        $CurrentVersion = $VersionLine.Substring($currentVersionIndex + 2)
+        $timestamp = Get-Date -Format yyyyMMddHHmm
+        Write-Output "version=$CurrentVersion" | Set-Content "$TempDir\backup.properties"
+        Write-Output "timestamp=$timestamp" | Add-Content "$TempDir\backup.properties"
+        #Write-Output "user=$OHUser" | Add-Content "$TempDir\backup.properties"
```

Comment:
Setting this to something like "windows" would be better for cross platform compatibility.

Reply:
What is the 'this' you are referring to? The directory, property, file, ...?

Reply:
Apologies, I assume you commented out this line because `OHUser` and `OHGroup` is blank? If you set to something like user=`openhab`, group=`windows` or even better the default: `user=openhab`, `group=openhab` then it would be clearer to the restore scripts on other operating systems.

Reply:
Yeah, in Windows there isn't a real reason to grab that info as there is in your case so I just left if commented out in case someone thought we needed something there. I'll just hardcode 'openhab' for both then.

Reply:
Case is important in unix/macos, so please let it be 'openhab'.